packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.slow-spec.ts

```diff
@@ -848,3 +930,109 @@ describe('ripgrep-search-in-workspace-server', function (): void {
+        [true, false].forEach(excludeFlag => {
+            const excludePrefix = excludeFlag ? '!' : '';
+            const filename = 'carrots';
+            const expected = [
+                `--glob=${excludePrefix}**/${filename}`,
+                `--glob=${excludePrefix}**/${filename}/*`
+            ];
+
+            const actual = ripgrepServer['filePatternToGlobs'](filename, rootDirA, excludeFlag);
+            expect(expected).to.have.deep.members(actual);
+        });
+    });
+
+    it('Resolve path to glob - glob prefixed folder', function (): void {
+        [true, false].forEach(excludeFlag => {
+            const excludePrefix = excludeFlag ? '!' : '';
+            const filename = 'carrots';
+            const inputPath = `**/${filename}/`;
+            const expected = [
+                `--glob=${excludePrefix}**/${filename}/`,
+                `--glob=${excludePrefix}**/${filename}/*`
+            ];
+
+            const actual = ripgrepServer['filePatternToGlobs'](inputPath, rootDirA, excludeFlag);
+            expect(expected).to.have.deep.members(actual);
+        });
+    });
+
+    it('Resolve path to glob - path segment', function (): void {
+        [true, false].forEach(excludeFlag => {
+            const excludePrefix = excludeFlag ? '!' : '';
+            const filename = 'carrots';
+            const inputPath = `/${filename}`;
+            const expected = [
+                `--glob=${excludePrefix}**/${filename}`,
+                `--glob=${excludePrefix}**/${filename}/*`
+            ];
+
+            const actual = ripgrepServer['filePatternToGlobs'](inputPath, rootDirA, excludeFlag);
+            expect(expected).to.have.deep.members(actual);
+        });
+    });
+
+    it('Resolve path to glob - already a glob', function (): void {
+        [true, false].forEach(excludeFlag => {
+            const excludePrefix = excludeFlag ? '!' : '';
+            const filename = 'carrots';
+            const inputPath = `${filename}/**/*`;
+            const expected = [
+                `--glob=${excludePrefix}**/${filename}/**/*`,
+            ];
+
+            const actual = ripgrepServer['filePatternToGlobs'](inputPath, rootDirA, excludeFlag);
+            expect(expected).to.have.deep.members(actual);
+        });
+    });
+
+    it('Resolve path to glob - path segment glob suffixed', function (): void {
+        [true, false].forEach(excludeFlag => {
+            const excludePrefix = excludeFlag ? '!' : '';
+            const filename = 'carrots';
+            const inputPath = `/${filename}/**/*`;
+            const expected = [
+                `--glob=${excludePrefix}**/${filename}/**/*`,
+            ];
+
+            const actual = ripgrepServer['filePatternToGlobs'](inputPath, rootDirA, excludeFlag);
+            expect(expected).to.have.deep.members(actual);
+        });
+    });
+});
+
+function matchPatternToPathResult(pattern: string, expectedPath: string): void {
```

Comment:
I'm not sure if the naming can be improved here, the method ultimately checks if the pattern matches the expected path.
It does not `match` the pattern in itself, but rather determines if it matches.

Reply:
Renamed to: checkResolvedPathForPattern